src/lib.rs

```diff
@@ -172,82 +172,100 @@
-#![recursion_limit = "128"]
 
 #[macro_use]
-extern crate error_chain;
+extern crate err_derive;
+pub type Result<T> = std::result::Result<T, Error>;
 
+#[derive(Error, Debug)]
+pub enum Error {
+    /// Invalid account name.
+    #[error(display = "Invalid account name")]
+    InvalidAccountName,
 
+    /// Invalid account password.
+    #[error(display = "Invalid account password")]
+    InvalidAccountPassword,
 
-pub use error_chain::ChainedError;
+    /// Invalid display name.
+    #[error(display = "Invalid display name")]
+    InvalidDisplayName,
 
-error_chain! {
-    errors {
-        /// Invalid account name.
-        InvalidAccountName {
-            description("Invalid account name")
-        }
-        /// Invalid account password.
-        InvalidAccountPassword {
-            description("Invalid account password")
-        }
-        /// Invalid display name.
-        InvalidDisplayName {
-            description("Invalid display name")
-        }
-        /// Invalid database name.
-        InvalidDatabaseName {
-            description("Invalid database name")
-        }
-        /// Invalid executable path.
-        InvalidExecutablePath {
-            description("Invalid executable path")
-        }
-        /// Invalid launch arguments.
-        InvalidLaunchArgument {
-            description("Invalid launch argument")
-        }
-        /// Invalid dependency name.
-        InvalidDependency {
-            description("Invalid dependency name")
-        }
-        /// Invalid machine name.
-        InvalidMachineName {
-            description("Invalid machine name")
-        }
-        /// Invalid service name.
-        InvalidServiceName {
-            description("Invalid service name")
-        }
-        /// Invalid start argument.
-        InvalidStartArgument {
-            description("Invalid start argument")
-        }
-        /// Invalid raw representation of [`ServiceState`].
-        InvalidServiceState(raw_value: u32) {
-            description("Invalid service state")
-            display("Invalid service state value: {}", raw_value)
-        }
-        /// Invalid raw representation of [`ServiceControl`].
-        InvalidServiceControl(raw_value: u32) {
-            description("Invalid service control")
-            display("Invalid service control value: {}", raw_value)
-        }
-        /// Invalid raw representation of [`ServiceStartType`].
-        InvalidServiceStartType(raw_value: u32) {
-            description("Invalid service start type")
-            display("Invalid service start type: {}", raw_value)
-        }
-        /// Invalid raw representation of [`ServiceErrorControl`].
-        InvalidServiceErrorControl(raw_value: u32) {
-            description("Invalid service error control type")
-            display("Invalid service error control type: {}", raw_value)
-        }
-    }
-    foreign_links {
-        System(::std::io::Error) #[doc = "System call error"];
```

Comment:
I did not check that we had lumped all `io::Error`s into one error before this PR. Sorry for picking about you just keeping the same style :/

Reply:
no worries, i figured you just wanted the enhancement